oaknational/Oak-Web-Application
# LESQ-1109: year group search filters

## Description

Music year: 1986

- Wire in changes to search api into OWA
- Plug in mv_search_page_10_0_0 into OWA - which should remove redundant search filters https://www.notion.so/oaknationalacademy/I-want-to-remove-redundant-subject-filters-on-search-11b26cc4e1b180d4a879c53ff2a06302
- Add year group search filters to search page
- Update active search filters when year group filter selected
- Update corresponding tests and schema definitions


## Issue(s)

Fixes #LESQ-1109

### AC

- [ ] Update search api to enable filtering by year
- [ ] In OWA update search to enable filtering by year
- [ ] If it is straightforward to exclude ‘Reception’ then do so, otherwise leave it (as we still get EYFS as a keystage anyway)
- [ ] Update the search result cards to display year

## How to test

1. Go to https://deploy-preview-2929--oak-web-application.netlify.thenational.academy/teachers/search
2. Click on any year group filter
3. You should see lessons and units have the associated year group with that filter

## Screenshots

How it should look:

https://github.com/user-attachments/assets/d0a3fe64-de39-46c8-9dba-ca8758fba689


## Checklist
